test/RequestChannelTest.cpp

```diff
@@ -88,11 +88,299 @@ TEST(RequestChannelTest, RequestOnDisconnectedClient) {
+      ->subscribe(requestSubscriber);
+
+  // make sure a few things have streamed
+  requestSubscriber->awaitValueCount(30);
+
+  worker.getEventBase()->runInEventBaseThread(
+      [=]() {
+        requestSubscriber->cancel();
+        requestSubscriber->request(50);
+      });
+
+  responderSubscriber->awaitTerminalEvent();
+  EXPECT_LT(requestSubscriber->getValueCount(), 100);
+
+  // Responder Subscriber should be at 100
+  // Requester Subscriber should be < 100
+  LOG(INFO) << "Responder Subscriber: " << responderSubscriber->getValueCount();
+  LOG(INFO) << "Requester Subscriber: " << requestSubscriber->getValueCount();
+}
+
+class TestChannelResponderFailure : public rsocket::RSocketResponder {
+ public:
+  TestChannelResponderFailure()
+      : testSubscriber_{TestSubscriber<std::string>::create()} {}
+
+  yarpl::Reference<Flowable<rsocket::Payload>> handleRequestChannel(
+      rsocket::Payload initialPayload,
+      yarpl::Reference<Flowable<rsocket::Payload>> requestStream,
+      rsocket::StreamId) override {
+    // add initial payload to testSubscriber values list
+    testSubscriber_->onNext(initialPayload.moveDataToString());
+
+    requestStream->map([](auto p) { return p.moveDataToString(); })
+        ->subscribe(testSubscriber_);
+
+    return Flowables::error<Payload>(
+        std::runtime_error("A wild Error appeared!"));
+  }
+
+  Reference<TestSubscriber<std::string>> getChannelSubscriber() {
+    return testSubscriber_;
+  }
+
+ private:
+  Reference<TestSubscriber<std::string>> testSubscriber_;
+};
+
+TEST(RequestChannelTest, FailureOnResponderRequesterSees) {
+  auto responder = std::make_shared<TestChannelResponderFailure>();
+
+  folly::ScopedEventBaseThread worker;
+  auto server = makeServer(responder);
+  auto client = makeClient(worker.getEventBase(), *server->listeningPort());
+  auto requester = client->getRequester();
+
+  auto requestSubscriber = TestSubscriber<std::string>::create();
+  auto responderSubscriber = responder->getChannelSubscriber();
+
+  int64_t requesterRangeEnd = 10;
+
+  auto requesterFlowable =
+      Flowables::range(1, requesterRangeEnd)->map([&](int64_t v) {
+        std::stringstream ss;
+        ss << "Requester stream: " << v << " of " << requesterRangeEnd;
+        std::string s = ss.str();
+        return Payload(s, "metadata");
+      });
+
+  requester->requestChannel(requesterFlowable)
+      ->map([](auto p) { return p.moveDataToString(); })
+      ->subscribe(requestSubscriber);
+
+  // failure streaming from Responder
+  requestSubscriber->awaitTerminalEvent();
+  requestSubscriber->assertOnErrorMessage("A wild Error appeared!");
+
+  responderSubscriber->awaitTerminalEvent();
+  responderSubscriber->assertValueAt(0, "Requester stream: 1 of 10");
+  responderSubscriber->assertValueAt(9, "Requester stream: 10 of 10");
+}
+
+TEST(RequestChannelTest, DISABLED_FailureOnRequestRequesterSees) {
+  // ???
+}
+
+TEST(RequestChannelTest, DISABLED_FailureFromRequester) {
```

Comment:
??

Reply:
The description for these test cases was:
- failure on request, requester sees
- failure from requester ... what happens?
I'm not clear on what this means.

Reply:
I don't know what it means either. Let's remove this then.